iceoryx_utils/include/iceoryx_utils/internal/cxx/storable_function.hpp

```diff
@@ -0,0 +1,165 @@
+// Copyright (c) 2020 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef IOX_UTILS_STORABLE_FUNCTION_HPP
+#define IOX_UTILS_STORABLE_FUNCTION_HPP
+
+#include "iceoryx_utils/cxx/function_ref.hpp"
+#include "iceoryx_utils/cxx/type_traits.hpp"
+#include "iceoryx_utils/internal/cxx/static_storage.hpp"
+
+#include <iostream>
+#include <type_traits>
+#include <utility>
+
+namespace iox
+{
+namespace cxx
+{
+template <typename ReturnType, typename... Args>
+using signature = ReturnType(Args...);
+
+template <typename StorageType, typename T>
+class storable_function;
+
+/// @brief A storable alternative of std::function which not necessarily uses dynamic memory.
```

Comment:
Yes it does, StorageType can essentially define how it will store the underlying object. And it can use dynamic memory...if StorageType does. It is very flexible.